hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -951,16 +1007,47 @@ bool MDMParser::powerOff(void)
         // Try to power off
         for (int i=0; i<3; i++) { // try 3 times
             if (!_atOk()) {
+                if (_dev.dev == DEV_SARA_R410) {
+                    check_ri = true;
+                    // If memory issue is present, ensure we don't force a power off too soon
+                    // to avoid hitting the 124 day memory housekeeping issue, AT+CPWROFF will
+                    // handle this delay automatically, or timeout after 40s.
+                    if (_memoryIssuePresent) {
+                        MDM_INFO("\r\n[ Modem::powerOff ] AT Failure, waiting up to 30s to power off with PWR_UC...");
+                        system_tick_t now = HAL_Timer_Get_Milli_Seconds();
+                        if (_timePowerOn == 0) {
+                            // fallback to max timeout of 30s to be safe
+                            _timePowerOn = now;
+                        }
+                        // check for timeout (VINT == LOW, Powered on 30s ago, Registered 20s ago)
+                        do {
+                            now = HAL_Timer_Get_Milli_Seconds();
+                            // prefer to timeout 20s after registration if we are registered
+                            if (_timeRegistered) {
+                                if (now - _timeRegistered >= 20000UL) {
+                                    break;
+                                }
+                            } else if (_timePowerOn && now - _timePowerOn >= 30000UL) {
+                                break;
+                            }
+                            HAL_Delay_Milliseconds(1); // just wait
```

Comment:
Let's increase this to something like `100`.

Reply:
Ok, done.